### DISCUSSION TITLE
[FR] MoveP for UR

### DISCUSSION THREAD

**Discussion by KonradJuenger** [visose/Robots#130] [Q&A] [answered]
Hi, 

as a new version is on the horizon, could you include a "moveP" mode for UR robots? MoveP allows for a constant speed profile, which is quite handy for anything where distance/time is critical e.g: 3D printing, drawing with inks, 

I've added this functionality in this [fork](https://github.com/KonradJuenger/Robots), but since it's a bit of a hack there is no PR. -- But it would be awesome if something like it could be part of the official release!
best

**Comment by visose** [comment]
I can add it to the todo list, but won't make it to the next release. I'll try to add support also to other manufacturers besides UR (like KUKA's C_VEL) so might take some time to get right.

For things that are distance/time critical as you say, I think trying to sync with the robot without real-time feedback is a fragile solution. For a robust solution feed the material using a servo as an external axis (I understand this won't be an option for budget constrained architecture schools 😄), or using an output that monitors the actual speed of the robot.

**Comment by visose** [comment, marked as answer]
MoveP support is now available for UR through the `Process` motion type on Cartesian targets. Add the Motion input on Create Target and set it to `Process`; the UR postprocessor emits `movep(...)`. Closing this as implemented.
